### PR TITLE
Integrate Cadene pretrainedmodels [fixes #142]

### LINKED ISSUE #142: Integrate Cadene pre-trained models
[Cadene](https://github.com/Cadene/pretrained-models.pytorch) is a great repository of pre-trained Convnets.  It's probably the largest single source of pre-trained models available for PyTorch. 

Many people are probably using Cadene and therefore it would be helpful to offer a simple bridge to Distiller, explaining how to use Cadene models from [compress_classifier.py](https://github.com/NervanaSystems/distiller/blob/master/examples/classifier_compression/compress_classifier.py).

Cheers
Neta

### PR DESCRIPTION
**Pull request by barrh**
Edited create_model() to import models from pretrainedmodels.
Added tests to the same function.

### CODE REVIEW

File: distiller/models/__init__.py [hunk 1]
```diff
@@ -20,6 +20,7 @@
 import torchvision.models as torch_models
 from . import cifar10 as cifar10_models
 from . import imagenet as imagenet_extra_models
+import pretrainedmodels
```
Comment (nzmora) [inline distiller/models/__init__.py]: I can't believe they named their package with such a generic name, and got away with it!

File: distiller/models/__init__.py [hunk 2]
```diff
@@ -62,29 +62,37 @@ def create_model(pretrained, dataset, arch, parallel=True, device_ids=None):
...
+            model = imagenet_extra_models.__dict__[arch]()
         elif arch in pretrainedmodels.model_names:
             model = pretrainedmodels.__dict__[arch](
                         pretrained=(dataset if pretrained else None))
         else:
-            assert not pretrained, "Model %s (ImageNet) does not have a pretrained model" % arch
-            model = imagenet_extra_models.__dict__[arch]()
+            error_message = ''
+            if arch not in IMAGENET_MODEL_NAMES:
+                error_message = "Model %s is not supported for dataset %s" % (arch, 'ImageNet')
```
Comment (nzmora) [inline distiller/models/__init__.py]: It's not that important, but we know that the dataset is ImageNet, so why pass it as a parameter to the string constructor (as in line 81)?

Comment (barrh) [reply distiller/models/__init__.py]: I agree. I guess I didn't give it much thought. I also think the info message on line $62 is probably redundant.
Ping me tomorrow if you want me refine to this.

**Comment by barrh** [comment]
Regarding your warning message when working when pretrainedmodels, please note I couldn't load 'nasnetalarge' with this patch. Moreover, the pretrained 'nasnetamoblie' (which does load) doesn't perform as expected out of the box.
I didn't have much time to look into that, but it looks like these models are WIP.

**Pull request merged**